summarize/src/signed_duration.rs

```diff
@@ -0,0 +1,118 @@
+use std::cmp::{Ord, Ordering, PartialOrd};
+use std::fmt;
+use std::ops::Sub;
+use std::time::Duration;
+use serde::{Deserialize, Serialize};
+
+#[derive(Serialize, Deserialize, Clone, Copy, Eq, PartialEq)]
+pub struct SignedDuration {
+    pub duration: Duration,
+    pub is_positive: bool,
+}
+
+impl From<Duration> for SignedDuration {
+    fn from(d: Duration) -> SignedDuration {
+        SignedDuration {
+            duration: d,
+            is_positive: true
+        }
+    }
+}
+
+impl Ord for SignedDuration {
+    fn cmp(&self, other: &SignedDuration) -> Ordering {
+        match (self.is_positive, other.is_positive) {
```

Comment:
Maybe easier to change this to `duration.as_nanos() as i128` and then using is_positive to flip the sign if needed?

Reply:
That's a really good idea! That simplifies this a lot.